script.js

```diff
@@ -1,25 +1,28 @@
 "use strict"
 
-const small_film_set = [
-    { id: 1, title: "The Shawshank Redemption", year: 1994, votes: 678790, rating: 9.2, rank: 1, category: "Thriller" },
-    { id: 2, title: "The Godfather", year: 1972, votes: 511495, rating: 9.2, rank: 2, category: "Crime" },
-    { id: 3, title: "The Godfather: Part II", year: 1974, votes: 319352, rating: 9.0, rank: 3, category: "Crime" },
-    { id: 4, title: "The Good, the Bad and the Ugly", year: 1966, votes: 213030, rating: 8.9, rank: 4, category: "Western" },
-    { id: 5, title: "Pulp fiction", year: 1994, votes: 533848, rating: 8.9, rank: 5, category: "Crime" },
-    { id: 6, title: "12 Angry Men", year: 1957, votes: 164558, rating: 8.9, rank: 6, category: "Western" }
-]
-
 function saveItem() {
-    const validationResult = $$("form").validate();
-    if (validationResult) {
-        const item_data = $$("form").getValues();
-        $$("table").add(item_data);
-        $$("form").clear();
+    const form = $$("form");
+    const table = $$("table");
+    const item_data = form.getValues();
+    if (item_data.id) {
+        table.updateItem(item_data.id, item_data);
```

Comment:
missing validation before update

Reply:
it should be done like this: 

- validation
- getting values and reading `id`
- add or update based on the existence of the id field

Reply:
There was an issue in tranformering ratings and votes values before verification. But now it's fixed.